src/nbl/video/CVKLogicalDevice.h

```diff
@@ -75,44 +185,137 @@ class CVKLogicalDevice final : public ILogicalDevice
+        {
+            if (_fences[i]->getAPIType() != EAT_VULKAN)
+                return;
+
+            vk_fences[i] = reinterpret_cast<CVulkanFence*>(_fences[i])->getInternalObject();
+        }
+
+        vkResetFences(m_vkdev, _count, vk_fences);
     }
             
-    IGPUFence::E_STATUS waitForFences(uint32_t _count, IGPUFence** _fences, bool _waitAll, uint64_t _timeout) override
+    IGPUFence::E_STATUS waitForFences(uint32_t _count, IGPUFence*const* _fences, bool _waitAll, uint64_t _timeout) override
     {
-        return IGPUFence::E_STATUS::ES_ERROR;
+        constexpr uint32_t MAX_FENCE_COUNT = 100u;
+
+        assert(_count <= MAX_FENCE_COUNT);
+
+        VkFence vk_fences[MAX_FENCE_COUNT];
+        for (uint32_t i = 0u; i < _count; ++i)
+        {
+            if (_fences[i]->getAPIType() != EAT_VULKAN)
+                return IGPUFence::E_STATUS::ES_ERROR;
+
+            vk_fences[i] = reinterpret_cast<CVulkanFence*>(_fences[i])->getInternalObject();
+        }
+
+        VkResult result = vkWaitForFences(m_vkdev, _count, vk_fences, _waitAll, _timeout);
+        switch (result)
+        {
+        case VK_SUCCESS:
+            return IGPUFence::ES_SUCCESS;
+        case VK_TIMEOUT:
+            return IGPUFence::ES_TIMEOUT;
+        default:
+            return IGPUFence::ES_ERROR;
+        }
     }
             
     const core::smart_refctd_dynamic_array<std::string> getSupportedGLSLExtensions() const override
     {
         return nullptr;
     }
             
-    core::smart_refctd_ptr<IGPUCommandPool> createCommandPool(uint32_t _familyIx, IGPUCommandPool::E_CREATE_FLAGS flags) override
+    core::smart_refctd_ptr<IGPUCommandPool> createCommandPool(uint32_t familyIndex, std::underlying_type_t<IGPUCommandPool::E_CREATE_FLAGS> flags) override
     {
-        return nullptr;
+        VkCommandPoolCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_COMMAND_POOL_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // pNext must be NULL
+        vk_createInfo.flags = static_cast<VkCommandPoolCreateFlags>(flags);
+        vk_createInfo.queueFamilyIndex = familyIndex;
+
+        VkCommandPool vk_commandPool = VK_NULL_HANDLE;
+        if (vkCreateCommandPool(m_vkdev, &vk_createInfo, nullptr, &vk_commandPool) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanCommandPool>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), flags, familyIndex, vk_commandPool);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
             
-    core::smart_refctd_ptr<IDescriptorPool> createDescriptorPool(IDescriptorPool::E_CREATE_FLAGS flags, uint32_t maxSets, uint32_t poolSizeCount, const IDescriptorPool::SDescriptorPoolSize* poolSizes) override
+    core::smart_refctd_ptr<IDescriptorPool> createDescriptorPool(
+        IDescriptorPool::E_CREATE_FLAGS flags, uint32_t maxSets, uint32_t poolSizeCount,
+        const IDescriptorPool::SDescriptorPoolSize* poolSizes) override
```

Comment:
@Crisspl does the OpenGL backend care about `maxSets` and `poolSizes` at all?

Reply:
opengl backend doesnt care descriptor pools at all